### ISSUE TITLE
Input dual hatch distinct function broken

### ISSUE THREAD

**Issue by DSRLIN** [GregTechCEu/GregTechCEu-1.19#1888]
### Checked for existing issues

- [X] I have checked for existing issues, and have found none.

### Tested latest version

- [X] I have checked that this occurs on the latest version.

### GregTech CEu Version

1.20.1-1.4.0-build_1059

### Minecraft Version

1.20.1

### Recipe Viewer Installed

JEI

### Environment

Singleplayer

### Cross-Mod Interaction

No

### Other Installed Mods

GTM Develop environment

### Expected Behavior

On a single multiblock, with both input dual hatches' distinct function on, item in hatch A should not react with fluid in hatch B.

### Actual Behavior

The reaction still happens.

### Steps to Reproduce

1. Put 2 input dual hatch on single multiblock machine.
2. Enable distinct function for both hatch.
3. Select 1 item and 1 fluid that can react with each other, e.g. sodium dust and chlorine in huge chemical reactor.
4. Put the item into hatch A and put the fluid into hatch B.
5. The reaction happens.

**Comment by wedgevic** [comment]
I can confirm, I ran into this as well

**Comment by Alehopper** [comment]
Still confirm with solidifire multiblock on latest version

**Comment by krossgg** [comment]
Known issue. Requires heavy rewrite of logic. WIP.

**Comment by domixxy** [comment]
I can verify that, same problem here

**Comment by krossgg** [comment]
Yes, thank you, this is still being worked on - see #2544. We are planning on it for 1.7.0
